src/fluree/db/flake/match.cljc

```diff
@@ -200,6 +201,50 @@
       (async/close! out-ch))
     out-ch))
 
+(defn match-property-flakes
+  [solution triple db property-flakes]
+  (map (fn [flake]
+         (where/match-flake solution triple db flake))
+       property-flakes))
+
+(defn match-subject-property
+  [initial-solutions triple-map db property-flakes]
+  (let [pid        (->> property-flakes first flake/p)
+        triple     (-> triple-map (get pid) first)]
+    (mapcat (fn [solution]
+              (match-property-flakes solution triple db property-flakes))
+            initial-solutions)))
+
+(defn match-join
+  [solution triples db join]
+  (let [triple-map (group-by (fn [[_ p-mch _]]
+                               (where/get-sid p-mch db))
+                             triples)]
+    (loop [[s-chunk & r] join
+           new-solutions [solution]]
+      (if s-chunk
+        (let [new-solutions* (match-subject-property new-solutions triple-map db s-chunk)]
+          (recur r new-solutions*))
+        new-solutions))))
+
+(defn match-properties
+  [db tracker solution triples error-ch]
+  (let [triples* (->> triples
+                      (map (fn [triple]
+                             (where/assign-matched-values triple solution)))
+                      (map (partial where/compute-sids db)))]
+    (if (every? some? triples*)
+      (let [property-ranges (->> triples*
+                                 (map (fn [triple]
+                                        (where/resolve-flake-range db tracker error-ch triple :psot)))
```

Comment:
Just checking to make sure I understand - this is the only circumstance where we will use the `:psot` index, right? At least, in the current implementation.